File: robocode-tankroyale-server/src/main/java/dev/robocode/tankroyale/server/events/BulletHitBulletEvent.java
package dev.robocode.tankroyale.server.events;

import lombok.Value;
import dev.robocode.tankroyale.server.model.Bullet;

/**
 * Event sent when a bullet hits another bullet.
 *
 * @author Flemming N. Larsen
 */
@Value
public class BulletHitBulletEvent implements Event {

	/** Turn number when event occurred */
	int turnNumber;

	/** Bullet that hit another bullet */
	Bullet bullet;

	/** Bullet that got hit by the bullet */
	Bullet hitBullet;
}

File: server/src/main/kotlin/dev/robocode/tankroyale/server/event/BulletHitBulletEvent.kt
package dev.robocode.tankroyale.server.event

import dev.robocode.tankroyale.server.model.IBullet

/** Event sent when a bullet hits another bullet. */
class BulletHitBulletEvent(
    /** Turn number when event occurred */
    override val turnNumber: Int,

    /** Bullet that hit another bullet */
    val bullet: IBullet,

    /** Bullet that got hit by the bullet */
    val hitBullet: IBullet,
) : Event()